internal/zinc-persist/src/main/scala/sbt/internal/inc/FileBasedStore.scala

```diff
@@ -7,17 +7,53 @@ package inc
 
 import java.io.File
 import sbt.io.{ IO, Using }
+import xsbti.SafeLazy
 import xsbti.compile.{ CompileAnalysis, MiniSetup }
+import xsbti.api.{ AnalyzedClass, Lazy }
 
 object FileBasedStore {
-  def apply(file: File): AnalysisStore = new AnalysisStore {
+  def apply(analysisFile: File, apiInfoFile: File): AnalysisStore =
+    new FileBasedStore0(analysisFile, apiInfoFile)
+
+  private final class FileBasedStore0(analysisFile: File, apiInfoFile: File) extends AnalysisStore {
+    val apiInfoStore = new FileBasedAPIInfoStore0(apiInfoFile)
     def set(analysis: CompileAnalysis, setup: MiniSetup): Unit = {
-      Using.fileWriter(IO.utf8)(file) { writer => TextAnalysisFormat.write(writer, analysis, setup) }
+      apiInfoStore.set(analysis match { case a: Analysis => a.apis })
+      Using.fileWriter(IO.utf8)(analysisFile) { writer => TextAnalysisFormat.write(writer, analysis, setup) }
     }
 
     def get(): Option[(CompileAnalysis, MiniSetup)] =
       try { Some(getUncaught()) } catch { case _: Exception => None }
+
     def getUncaught(): (CompileAnalysis, MiniSetup) =
-      Using.fileReader(IO.utf8)(file) { reader => TextAnalysisFormat.read(reader) }
+      {
+        val (analysis0: Analysis, setup) = Using.fileReader(IO.utf8)(analysisFile) { reader => TextAnalysisFormat.read(reader) }
```

Comment:
I think all this logic should be moved to  APIsF's read and write methods (in TextAnalysisFormat.scala) not to construct twice AnalyzedClass instance.